parsers/thueringen.py

```diff
@@ -0,0 +1,199 @@
+	return datetime.strptime(start_date_str.group(), '%d.%m.%Y')
+
+
+def parse_fees(document):
+	fees_regex = re.compile('Bedienstete.*')
+
+	fees = document.find_all('p', class_='MsoNormal', string=fees_regex)
+
+	if len(fees) != 1:
+		return None, None
+
+	fees = fees[0]
+
+	fee_strings = fees_regex.findall(fees.text)
+	for amount_candidate in fee_strings:
+		amount_strings = amount_regex.findall(amount_candidate)
+		if len(amount_strings) != 2:
+			continue
+
+		employees_fee = float(amount_strings[0].replace(',', '.'))
+		guests_fee = float(amount_strings[1].replace(',', '.'))
+		return employees_fee, guests_fee
+
+	return None, None
+
+
+def parse_ingredients(document):
+	contextbox = document.find('div', class_='kontextbox')
+
+	shorts = contextbox.find_all('p')
+
+	digits_regex = re.compile('(\d)+: .*')
+	chars_regex = re.compile('(\w)+: .*')
+
+	digits = dict()
+	chars = dict()
+
+	if len(shorts) < 3:
+		return
+	for short in shorts[1].childGenerator():
+		if digits_regex.search(str(short)):
+			n = str(short).strip('\r\n').split(':')
+			digits[n[0]] = n[1]
+
+	for short in shorts[2].childGenerator():
+		if chars_regex.search(str(short)):
+			c = str(short).strip('\r\n').split(':')
+			chars[c[0]] = c[1]
+
+	shorts = contextbox.find_all('td')
+
+	if len(shorts) >= 1:
+		for short in shorts[1].childGenerator():
+			if chars_regex.search(str(short)):
+				c = str(short).strip('\r\n').split(':')
+				chars[c[0]] = c[1]
+
+	return digits, chars
+
+
+def ingredient_list_to_notes(ingredients_list, chars, digits):
+	ingredients = []
+	for ingredient in ingredients_list:
+
+		if digits.get(ingredient) is not None:
+			ingredients.append(digits[ingredient][1:])
+
+		if chars.get(ingredient) is not None:
+			ingredients.append(chars[ingredient][1:])
+
+	return ingredients
+
+
+def parse_meals(day, digits, chars):
+	ingredients_regex = re.compile('Inhalt:.*')
+
+	meals_t_rows = day.find_all('tr')
+
+	for meal_data in meals_t_rows:
+		meal_t_datas = meal_data.find_all('td')
+		if len(meal_t_datas) != 3:
+			continue
+
+		category = [meal_t_datas[0].text]
+		if len(category) == 1:
+			category = category[0]
+		else:
+			category = 'Hauptgerichte'
+
+		notes = []
+		ingredients = ingredients_regex.findall(meal_t_datas[1].text)
+
+		if len(ingredients) > 0:
+			ingredients_list = ingredients[0].strip('Inhalt: ').strip('\r').split(',')
+
+			notes = ingredient_list_to_notes(ingredients_list, chars, digits)
+
+		main_dish = ingredients_regex.sub('', meal_t_datas[1].text).strip()
+
+		students_fee_string = amount_regex.findall(meal_t_datas[2].text)
+		if len(students_fee_string) != 1:
+			continue
+
+		students_fee = float(students_fee_string[0].replace(',', '.'))
+		costs = {'student': students_fee}
+
+		yield (main_dish, notes, costs, category)
+
+
+def parse_url(url, today=False):
+	year = datetime.now().isocalendar()[0]
+	week = datetime.now().isocalendar()[1]
+
+	days_regex = re.compile('day_\d$')
+
+	canteen = LazyBuilder()
+
+	for i in range(0, 3):
```

Comment:
You should probably only fetch the current week when the today parser is requested.